participant/models.py

```diff
@@ -74,6 +76,31 @@ def actions(self):
 
         return actions
 
+    def phishing_email_behaviour_and_actions(self, email_id):
+        """
+        Aggregates actions related to a phishing email and determines behaviour.
+
+        :param email_id: UUID - id of an email
+        :return: (STRING, LIST) - returns a tuple of participant behaviour and actions
+        """
+
+        def participant_behaviour(actions):
+            pb = ParticipantBehaviour.NEUTRAL
+            if [a for a in actions if a.get('action_type') in POSITIVE_ACTIONS]:
+                pb = ParticipantBehaviour.POSITIVE
+            if [a for a in actions if a.get('action_type') in NEGATIVE_ACTIONS]:
+                pb = ParticipantBehaviour.NEGATIVE
+            return pb
+
+        actions = []
+
+        if email_id in self.exercise.phishing_email_ids:
+            for action_id, action_details in self.actions.items():
+                if email_id == action_details.get('email_id'):
```

Comment:
Reference to the issue https://github.com/cybsafe/phishtray/issues/221. I wonder if it is filtering the "top actions only".

Reply:
good point - the first action that the user made should be the first item in the list
we went through the specs with Ben yesterday and found that we won't need to separately send the first item as the FE guys can just take the first item from the array

note - if we decide to send the first item separately then we can look at introducing some logic that looks at the `timedeltas` of each item and extract it that way

Reply:
This is why I made a separate property only for the first ones. Anyways with this resolved no point in reviewing that PR now.